test/extended/testdata/gssapi/fedora/base/gssapi-tests.sh

```diff
@@ -0,0 +1,283 @@
+    os::cmd::expect_failure_and_not_text 'oc whoami' 'user1'
+
+    os::cmd::expect_failure_and_text 'oc login -u user2 -p wrongpassword' 'An invalid name was supplied'
+    os::cmd::expect_failure_and_not_text 'oc whoami' 'user2'
+
+    os::cmd::expect_failure_and_text 'oc login -u user2 -p password' 'An invalid name was supplied'
+    os::cmd::expect_failure_and_not_text 'oc whoami' 'user2'
+
+    os::cmd::expect_failure_and_text "oc login -u user3@${REALM}" "Can't find client principal user3@${REALM} in cache collection"
+    os::cmd::expect_failure_and_not_text 'oc whoami' 'user3'
+
+    os::cmd::expect_failure_and_text "oc login -u user4@${REALM} -p wrongpassword" "Can't find client principal user4@${REALM} in cache collection"
+    os::cmd::expect_failure_and_not_text 'oc whoami' 'user4'
+
+    os::cmd::expect_failure_and_text "oc login -u user5@${REALM} -p password" "Can't find client principal user5@${REALM} in cache collection"
+    os::cmd::expect_failure_and_not_text 'oc whoami' 'user5'
+
+fi
+
+# Client has unconfigured GSSAPI libs and server is GSSAPI with Basic fallback
+# Only BASIC works
+# Errors do NOT mention Kerberos
+
+if [[ ( "${CLIENT}" = 'CLIENT_MISSING_LIBS' || "${CLIENT}" = 'CLIENT_HAS_LIBS' ) && "${SERVER}" = 'SERVER_GSSAPI_BASIC_FALLBACK' ]]; then
+
+    os::cmd::expect_failure_and_text 'oc login <<< \n' 'Login failed \(401 Unauthorized\)'
+    os::cmd::expect_failure_and_text 'oc whoami' 'system:anonymous'
+
+    os::cmd::expect_failure_and_text 'oc login -u user1 <<EOF
+    EOF' 'Login failed \(401 Unauthorized\)'
+    os::cmd::expect_failure_and_not_text 'oc whoami' 'user1'
+
+    os::cmd::expect_failure_and_text 'oc login -u user2 -p wrongpassword' 'Login failed \(401 Unauthorized\)'
+    os::cmd::expect_failure_and_not_text 'oc whoami' 'user2'
+
+    os::cmd::expect_success_and_text 'oc login -u user2 -p password' 'Login successful.'
+    os::cmd::expect_success_and_text 'oc whoami' "user2@${REALM}"
+    os::cmd::expect_success_and_text 'oc logout' "user2@${REALM}"
+    os::cmd::try_until_text 'oc whoami' 'system:anonymous' # Make sure token is gone
+
+    os::cmd::expect_failure_and_text "oc login -u user3@${REALM} <<EOF
+    EOF" 'Login failed \(401 Unauthorized\)'
+    os::cmd::expect_failure_and_not_text 'oc whoami' 'user3'
+
+    os::cmd::expect_failure_and_text "oc login -u user4@${REALM} -p wrongpassword" 'Login failed \(401 Unauthorized\)'
+    os::cmd::expect_failure_and_not_text 'oc whoami' 'user4'
+
+    os::cmd::expect_success_and_text "oc login -u user5@${REALM} -p password" 'Login successful.'
+    os::cmd::expect_success_and_text 'oc whoami' "user5@${REALM}"
+    os::cmd::expect_success_and_text 'oc logout' "user5@${REALM}"
+
+fi
+
+# Client has GSSAPI configured and server is GSSAPI only
+# Only GSSAPI works
+# Errors mention Kerberos
+
+if [[ "${CLIENT}" = 'CLIENT_HAS_LIBS_IS_CONFIGURED' && "${SERVER}" = 'SERVER_GSSAPI_ONLY' ]]; then
```

Comment:
It would seem like each one of these stanzas should merit a jUnit suite -- the suites are nestable